component.go

```diff
@@ -116,8 +117,15 @@ type component struct {
 var _ Instance = &componentImpl{}
 
 // Main is interface implemented by an application's main component.
-// This component is created automatically by `weaver.Run`.
-type Main interface{}
+// This component is instantiated and its Main() method called by
+// `weaver.Run`.
+type Main interface {
+	// TODO(reviewer): Should we make this method optional?
```

Comment:
If someone opts to not implement the `Main` method, what would be the behavior of `weaver.Run`? I guess we would still run the main component's `Init` method and then block forever? 

I like making `weaver.Ref[weaver.Main]` not compile somehow.

Reply:
Yes, my intention was that it would block.

I  will try to make weaver.Ref[weaver.Main] not compile. And perhaps make the code generator refuse to make RPCs for Main?